### PR TITLE
Case 29492:  AMP upgrade

### PR DESCRIPTION
**Pull request by dakala**

### CODE REVIEW

File: src/Pass/ImgTagDimensionsPass.php [hunk 1]
```diff
@@ -39,6 +50,33 @@ public function pass() {
...
+
+    foreach ($els as $el) {
+      if (!$el->hasAttr('src')) {
+        $el->remove();
+        continue;
+      }
+
+      $file_path = trim(parse_url($el->attr('src'), PHP_URL_PATH), '/');
+      if (!is_file($file_path)) {
+        $el->remove();
```
Comment (davidgrayston) [inline src/Pass/ImgTagDimensionsPass.php]: We need to be careful we don’t remove img tags with remote src

File: src/Pass/ImgTagDimensionsPass.php [hunk 2]
```diff
@@ -39,6 +50,33 @@ public function pass() {
           $el->attr('height', $info['height']);
         }
       }
+      else {
+        // Invalid file path.
+        $el->remove();
```
Comment (davidgrayston) [inline src/Pass/ImgTagDimensionsPass.php]: Could remove img tag with remote src

**Comment by davidgrayston** [comment]
Perhaps we could add a new processor that removes img tags after all other processors have run? New class that’s  appended to passes array?

**Comment by davidgrayston** [comment]
I’d leave this processor to only deal with img dimensions. New class should be created for other processing

**Comment by tashaharrison** [comment]
The new class removes the logo. This seems a bit aggressive to just remove all images which appear wrong. In most cases the improvements in 7.x-1.14 should fix the image issues. One of the issues when I'm testing it on VM is that on some of the articles the files don't exist locally, which may have been why it appeared the fixes from 1.14 weren't working.

**Comment by dakala** [comment]
@tashaharrison 

> This seems a bit aggressive to just remove all images which appear wrong.

Two types of tags fail AMP validation - empty image tags `` and tags without `height` and `width` attributes (because the `src` is invalid). That's why they're removed.

I haven't noticed the logo removal. Perhaps you can demo that to me please?

File: src/Pass/ImgTagDimensionsPass.php [hunk 3]
```diff
@@ -31,6 +31,9 @@ public function pass() {
       if (!$el->hasAttr('src')) {
         continue;
       }
+      if ($el->hasAttr('width') && $el->hasAttr('height')) {
+        return;
```
Comment (peteratdennis) [inline src/Pass/ImgTagDimensionsPass.php]: Don't return, continue

**Comment by jon-edery** [comment]
the following ticket is required to progress this issue: https://didev.manuscript.com/f/cases/29695/

**Pull request merged**